### DISCUSSION TITLE
Startup config template -- a feature or a bug?

### DISCUSSION THREAD

**Discussion by networkop** [srl-labs/containerlab#981] [Q&A] [answered]
I've just realised that due to [this](https://github.com/srl-labs/containerlab/blob/4fbe732d91753baf0ede4cdff98af39a9dc7db72/nodes/ceos/ceos.go#L163), cEOS startup configuration gets passed through the templating engine, allowing me to embed the templating logic in my startup config like: 
https://github.com/PacktPublishing/Network-Automation-with-Go/blob/main/topo-full/workdir/ceos.startup#L42-L46

Since it was not obvious from the documentation, I was wondering if this was done by design or is this a side effect and may be removed in the future?

**Comment by hellt** [comment, marked as answer]
Hi @networkop 
this is by design, yes.
You can also use gomplate functions there as well
